#! /bin/sh /usr/share/dpatch/dpatch-run
# Description: Fix vboxvideo_drm crash with kernel 2.6.33 / backported drm code
# Bug: http://www.virtualbox.org/ticket/6198
# Bug-Ubuntu: https://bugs.launchpad.net/bugs/535297

@DPATCH@

Index: trunk/src/VBox/Additions/linux/drm/vboxvideo_drm.c
===================================================================
--- trunk/src/VBox/Additions/linux/drm/vboxvideo_drm.c (revision 25591)
+++ trunk/src/VBox/Additions/linux/drm/vboxvideo_drm.c (revision 27248)
@@ -88,5 +88,11 @@
 		 .open = drm_open,
 		 .release = drm_release,
+                 /* This was changed with Linux 2.6.33 but Fedora backported this
+                  * change to their 2.6.32 kernel. */
+#if defined(DRM_UNLOCKED) || LINUX_VERSION_CODE >= KERNEL_VERSION (2, 6, 33)
+		 .unlocked_ioctl = drm_ioctl,
+#else
 		 .ioctl = drm_ioctl,
+#endif
 		 .mmap = drm_mmap,
 		 .poll = drm_poll,
